cloud/ec2.go

```diff
@@ -629,7 +629,14 @@ func (m *ec2Manager) GetInstanceStatuses(ctx context.Context, hosts []host.Host)
 			reservationsMap[*out.Reservations[i].Instances[0].InstanceId] = out.Reservations[i].Instances[0]
 		}
 		for i := range hostsToCheck {
-			status := ec2StatusToEvergreenStatus(*reservationsMap[*hostsToCheck[i]].State.Name)
+			instance, ok := reservationsMap[*hostsToCheck[i]]
+			if !ok {
+				return nil, errors.Errorf("host '%s' not included in DescribeInstances response", *hostsToCheck[i])
+			}
+			if instance.State == nil {
+				return nil, errors.Errorf("host '%s' has no state in DescribeInstances response", *hostsToCheck[i])
+			}
+			status := ec2StatusToEvergreenStatus(*instance.State.Name)
```

Comment:
maybe we also should check that `instance.State.Name != nil` as well as each of the fields referenced by `cacheHostData` (e.g. `instance.Placement`, `instance.Placement.AvailabilityZone`, `instance.LaunchTime`, etc.)?

Reply:
Done (in cacheHostData).